llvm/lib/Target/RISCV/RISCVInsertVSETVLI.cpp

```diff
@@ -538,6 +546,18 @@ DemandedFields getDemanded(const MachineInstr &MI, const RISCVSubtarget *ST) {
     Res.MaskPolicy = false;
   }
 
+  if (isVExtractInstr(MI)) {
+    assert(!RISCVII::hasVLOp(TSFlags));
+    // TODO: LMUL can be any larger value (without cost)
+    Res.TailPolicy = false;
+    Res.MaskPolicy = false;
+  }
+
+  if (isVInsertInstr(MI)) {
+    // TODO: LMUL can be any larger value (without cost)
+    Res.MaskPolicy = false;
```

Comment:
Setting `MaskPolicy` to false should have already been done by the earlier check for `!RISCVII::usesMaskPolicy(TSFlags)`.

That's true for most of the existing places we write MaskPolicy to false. So I'm not sure if this is an intentional choice we've made to repeat it.

Reply:
I don't think it is.  I'm going to drop it here, and then post a separate patch to cleanup the others.